packages/eslint-plugin-svelte/src/rules/derived-has-same-inputs-outputs.ts

```diff
@@ -1,7 +1,35 @@
 import type { TSESTree } from '@typescript-eslint/types';
+import type { Variable } from '@typescript-eslint/scope-manager';
 import { createRule } from '../utils/index.js';
 import type { RuleContext } from '../types.js';
 import { extractStoreReferences } from './reference-helpers/svelte-store.js';
+import { getScope } from '../utils/ast-utils.js';
+
+function findVariableForName(
+	context: RuleContext,
+	node: TSESTree.Node,
+	name: string,
+	expectedName: string
+): { hasConflict: boolean; variable: Variable | null } {
+	const scope = getScope(context, node);
+	let hasConflict = false;
+	let variable: Variable | null = null;
+
+	for (const v of scope.variables) {
+		if (hasConflict && variable) {
+			break;
+		}
+		if (v.name === expectedName) {
+			hasConflict = true;
```

Comment:
I think we need to check parent/global scopes also to check the conflict.

Reply:
why so?

the param is local to the function body. if the user chose to have a variable named the same in the outer scope, it won't conflict, it'll just cause shadowing (which they should be picking up via the shadowing rule in eslint)

Reply:
I didn't run tests but I guess...

```js
const $a = 1;
derived(a, (b) => {
  // output 1.
  console.log($a);
});
```

This code will be...

```js
const $a = 1;
derived(a, ($a) => {
  // output a value.
  console.log($a);
});
```

So I think the behavior will change in a way that’s hard to notice and could cause issues.

Reply:
you're right. this should be fixed now and i've added a test